pkg/recipes/terraform/module.go

```diff
@@ -49,6 +53,38 @@ type moduleInspectResult struct {
 	// Any other module information required in the future can be added here.
 }
 
+// downloadAndInspect handles downloading the TF module and retrieving the necessary information
+func downloadAndInspect(ctx context.Context, tf *tfexec.Terraform, options Options) (*moduleInspectResult, error) {
```

Comment:
This is copied over from https://github.com/radius-project/radius/blob/main/pkg/recipes/terraform/execute.go#L308, with main change being directly calling `tf.Get` instead of another function `downloadModule` calling `tf.Get`.

Reply:
Why did we need to move this function to a new location?

Reply:
I don't know how this ended up being in execute as the code evolved, but same reason why inspect module is in `module.go`. Abstracting out downloading and reading module into a separate file for maintainability. Let me know if you have any concerns that I might have missed.